### ISSUE TITLE
Use annotations to skip mutations

### ISSUE THREAD

**Issue by MercurieVV** [stryker-mutator/stryker4s#495]
### Summary
It would be cool, if I could disable some lines of code from some checking.
For example when I configure my HTTP client logging option with booleans.
```  private val client = RequestLogger(logHeaders = false, logBody = false)(ResponseLogger(logHeaders = false, logBody = true)(httpClient))```
I dont need to extract values to config. I also dont want to disable whole file from testing. One thing I need - disable this exact code from boolean inversion checking.

#### My proposal
If its possible, we could use java.lang.SuppressWarnings annotation. For example before code line put `SuppressWarnings(Array("stryker4s.extension.mutationtype.BooleanLiteral"))`
Then, engine could skip this line from checking

**Comment by hugo-vrijswijk** [comment]
Thanks for this feature request! Though I personally don't like adding test annotations to production code, I understand many people would find this useful. We should give any mutations skipped by this the [ignored](https://github.com/stryker-mutator/mutation-testing-elements/blob/master/packages/mutation-testing-metrics/src/api/Metrics.ts#L31-L34) status so they show up in the mutation testing report. 

If anyone else is interested in implementing this, there is currently no infrastructure for ignored mutants, or having mutants end up in the report other than by testing them, so that would likely require a bit of work

**Comment by MercurieVV** [comment]
I could try to implement it by myself, but I never worked with scalameta before. So I need read a bit before starting.

**Comment by MercurieVV** [comment]
Regarding "ignored" status.
Here I see statuses mapping code: 
https://github.com/stryker-mutator/stryker4s/blob/a10a3edd9d2f237c741ea9faa3596ac6157282f6/core/src/main/scala/stryker4s/report/mapper/MutantRunResultMapper.scala#L49-L56
And "ignored" is not used here. On stryker4s side "ignored" status even dont exist yet as you can see it from:
https://github.com/stryker-mutator/stryker4s/blob/a10a3edd9d2f237c741ea9faa3596ac6157282f6/core/src/main/scala/stryker4s/model/MutantRunResult.scala
I suppose it should be another task to create this status. Probably I could implement it.
